utils/make-report.py

```diff
@@ -16,170 +16,387 @@
         print("Error:", msg, "\n")
-    print("make-report -[hs] -i <file_log> [-d <num>] [-p <op regexp>]")
+    print("make-report.py -[svh] [-c <compile_log>] [-r <run_log>] [-l <num>] [-p <op regexp>]")
     print("")
-    print("Usage: scan onnx op report generated by `onnx-mlir1 with `--onnx-op-report`")
-    print("equal to `Simd` or `Parallel` and report statistics about the ops found")
-    print("in the provided input log.")
+    print("Usage: Report statistics on compiler and runtime characteristics of onnx ops.")
     print("")
-    print("Parameters:")
-    print("  -d/--detail <num>:     Print statistics:")
-    print("                         0: Just count successful/unsuccessful ops.")
-    print("                         1: Also count reasons for success/failure.")
-    print("                         2: Also list metrics.")
-    print("                         3: Also list node name.")
-    print("  -h/--help:             Print usage.")
-    print("  -i/--input <file_log>: Input file to be scanned.")
-    print("  -p/--pattern <regexp>: Focus only on ops that match the regexp pattern.")
-    print("  -s/supported:          Focus only on ops that are supported, i.e. skip unsupported ops.")
+    print("Compile-time statistics are collected from a `onnx-mlir` compiler output")
+    print("with the `--onnx-op-report` option equal to `Simd` or other supported sub-options.")
+    print("")
+    print("Runtime statistics are collected from the runtime output of a model compiled.")
+    print("with the `--profile-ir` option equal to `Onnx` or other supported sub-options.")
+    print("")
+    print("When both compile time and runtime statistics are provided at the same time,")
+    print("it will correlate the performance metrics with data gathered at compile time.")
     print("")
-    print("More info:")
+    print("Additional help.")
     print("  If you need more specific info on individual success/failure, run ")
     print("  `onnx-mlir --debug-only=lowering-to-krnl` and look at the compiler output.")
-    print("  Use `-d 3` to correlate the node name printed here with compiler output.")
+    print("  Use `-l 3` to correlate the node name printed here with compiler output.")
+    print("")
+    print("Parameters:")
+    print("  -c/--compile <file_log>: File name containing the compile time statistics.")
+    print("  -r/--runtime <file_log>: File name containing the runtime statistics.")
+    print("  -l/--level <num>:        Print statistics:")
+    print("                           0: Just count successful/unsuccessful ops.")
+    print("                           1: Also count reasons for success/failure.")
+    print("                           2: Also list metrics.")
+    print("                           3: Also list node name.")
+    print("  -f/--focus <regexp>:     Focus only on ops that match the regexp pattern.")
+    print("  -s/supported:            Focus only on ops that are supported, i.e. skip unsupported ops.")
```

Comment:
Could you explain more about supported/unsupported ops to make them clear? I guess they are related to SIMD and parallel.

Reply:
great suggestion
```
  -s/supported:        Focus only on ops that are supported. Namely, the report
                       will skip ops for which compile-time statistics list
                       the 'unsupported' keyword in its printout.
                       For SIMD/parallel statistics, this include all ops that
                       have currently no support for it.
```